### PR TITLE
Use calldata globals instead of syscalls

### PR DESCRIPTION
**Pull request by Danikim01**
closes #269 
This PR removes the use syscalls in calldataload opcode

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -952,6 +952,168 @@ pub(crate) fn compute_memory_cost<'c>(
...
+    let calldata_size_ptr = block
+        .append_operation(llvm_mlir::addressof(
+            context,
+            CALLDATA_SIZE_GLOBAL,
+            ptr_type,
+            location,
+        ))
+        .result(0)?;
+
+    // let zero = block
```
Comment (JulianVentura) [inline src/utils.rs:1088]: Here you forgot some commented code. Please, delete it.

File: src/utils.rs [hunk 2]
```diff
@@ -952,6 +952,168 @@ pub(crate) fn compute_memory_cost<'c>(
...
+
+    block.append_operation(llvm::store(
+        context,
+        calldata_size_syscall,
+        calldata_size_ptr.into(),
+        location,
+        LoadStoreOptions::default(),
+    ));
+
+    //Ok(calldata_size_syscall)
```
Comment (JulianVentura) [inline src/utils.rs:1113]: More commented code.

File: src/utils.rs [hunk 3]
```diff
@@ -952,6 +952,168 @@ pub(crate) fn compute_memory_cost<'c>(
...
+}
+
+pub(crate) fn setup_calldata_size<'c>(
+    op_ctx: &'c OperationCtx,
+    block: &'c Block,
+    location: Location<'c>,
+) -> Result<(), CodegenError> {
+    let context = op_ctx.mlir_context;
+    let calldata_size_syscall = op_ctx.get_calldata_size_syscall(block, location)?;
+    //let uint32 = IntegerType::new(context, 32).into();
```
Comment (JulianVentura) [inline src/utils.rs:1076]: More commented code

**Review by TomasArrachea — COMMENTED** [review]
LGTM

File: src/utils.rs [hunk 4]
```diff
@@ -1020,6 +1020,149 @@ pub(crate) fn compute_memory_cost<'c>(
...
+        calldata_ptr_syscall,
+        calldata_ptr_ptr.into(),
+        location,
+        LoadStoreOptions::default(),
+    ));
+
+    Ok(())
+}
+
+pub(crate) fn return_calldata_ptr<'c>(
```
Comment (MegaRedHand) [inline src/utils.rs:1075]: We should rename these `return_*` functions to `get_*` to avoid confusion

File: src/utils.rs [hunk 5]
```diff
@@ -1020,6 +1020,149 @@ pub(crate) fn compute_memory_cost<'c>(
...
+    let res = body.append_operation(llvm_mlir::global(
+        context,
+        CALLDATA_SIZE_GLOBAL,
+        uint32,
+        location,
+    ));
+    assert!(res.verify());
+}
+
+pub(crate) fn setup_calldata_ptr<'c>(
```
Comment (MegaRedHand) [inline src/utils.rs:1045]: These `setup_*` functions are used only once when initializing the `OperationCtx`. We should move them there

File: src/codegen/context.rs [hunk 6]
```diff
@@ -61,7 +61,6 @@ impl<'c> OperationCtx<'c> {
         // Append setup code to be run at the start
         generate_stack_setup_code(context, module, setup_block)?;
         generate_memory_setup_code(context, module, setup_block)?;
-        generate_calldata_setup_code(context, module, setup_block)?;
```
Comment (MegaRedHand) [inline src/codegen/context.rs:64]: This was OK here. Just pass `syscall_ctx` as parameter.

File: src/codegen/context.rs [hunk 7]
```diff
@@ -333,61 +332,6 @@ fn generate_memory_setup_code<'c>(
     Ok(())
 }
 
-fn generate_calldata_setup_code<'c>(
```
Comment (MegaRedHand) [inline src/codegen/context.rs:336]: This was mostly OK. It just needed to use the correct values.

**Review by MegaRedHand — CHANGES_REQUESTED** [review]
Left some comments

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**